src/Swift.Runtime/src/Swift/Data.cs

```diff
@@ -1,10 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Swift;
+using Swift.Runtime;
+
+
 /// <summary>
-/// Represents Foundation.Data type.
+/// Represents Swift Foundation.DataProtocol in C#.
+/// </summary>
+public interface ISwiftDataProtocol { }
+
+/// <summary>
+/// Represents Swift Foundation.ContiguousBytes in C#.
 /// </summary>
+public interface ISwiftContiguousBytes { }
 
-// TODO: https://github.com/dotnet/runtimelab/issues/2992
-public struct Data
+/// <summary>
+/// Represents Foundation.Data type.
+/// </summary>
+public struct Data : ISwiftObject
 {
     private long _flags;
     private IntPtr _object;
+
+    private static nuint _payloadSize = SwiftObjectHelper<Data>.GetTypeMetadata().Size;
```

Comment:
Since I'm nit picking and posting unrelated comments: This is another "startup cost" - this leads to several things:
- It gets the metadata (PInvoke)
- caches it
- Gets the size (another PInvoke)

And all of this will happen the first time anything just touches the Data type. While in reality we only need this when the type is instantiated (or maybe even later on).

Additionally I just noticed that this doesn't seem to be used anywhere... ???

Reply:
We should only add this when SwiftIndirectResult is used, since it appears to be the only case. I've added this to. Added to https://github.com/dotnet/runtimelab/issues/2990.